src/test/java/lambda/part2/exercise/Exercise2.java

```diff
@@ -0,0 +1,100 @@
+package lambda.part2.exercise;
+
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.function.Predicate;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void personHasNotEmptyLastNameAndFirstName() {
+        // TODO предикат Person -> boolean, проверяющий что имя и фамилия человека не пусты
+        Predicate<Person> validate = (person) -> !person.getFirstName().equals("") && !person.getLastName().equals("");
```

Comment:
There is good method ```string.isEmpty()``` which should have been recommended by IDEA here (by highlighting).
This is clearer than ```equals("")```